install_config/persistent_storage/persistent_storage_csi.adoc

```diff
@@ -0,0 +1,371 @@
+====
+
+[NOTE]
+====
+The external attacher must also run for CSI drivers that do not support third-party attach/detach operations. The external attacher will not issue any `ControllerPublish` or `ControllerUnpublish` operations to CSI driver. However, it still must run to implement the necessary {product-title} attachment API.
+====
+
+=== CSI Driver DaemonSet
+
+Finally, the CSI driver DaemonSet runs a pod on every node that allows {product-title} to mount storage provided by CSI driver to the node and use it in user workloads (pods) as persistent volumes (PVs). The pod with the CSI driver installed contains the following containers:
+
+* CSI driver registrar, which registers the CSI driver into the `openshift-node` service running on the node. The `openshift-node` process running on the node then directly connects with the CSI driver using the UNIX Domain Socket available on the node.
+* CSI driver.
+
+The CSI driver deployed on the node should have as few credentials to the storage backend as possible. {product-title} will only use the node plug-in set of CSI calls such as `NodePublish`/`NodeUnpublish` and `NodeStage`/`NodeUnstage` (if implemented).
+
+[[install-config-persistent-storage-csi-example-deployment]]
+== Example Deployment
+
+Since {product-title} does not ship with any CSI driver installed, this example shows how to deploy a community driver for OpenStack Cinder in {product-title}.
+
+. Create a new project where the CSI components will run and a new service account that will run the components. Explicit node selector is used to run the Daemonset with the CSI driver also on master nodes.
++
+[source,shell]
+----
+# oc adm new-project csi --node-selector=""
+Now using project "csi" on server "https://example.com:8443".
+
+# oc create serviceaccount cinder-csi
+serviceaccount "cinder-csi" created
+
+# oc adm policy add-scc-to-user privileged system:serviceaccount:csi:cinder-csi
+scc "privileged" added to: ["system:serviceaccount:csi:cinder-csi"]
+----
+
+. Apply this YAML file to create the Deployment with the external CSI attacher and provisioner, and DaemonSet with the CSI driver.
++
+[source,yaml]
+----
+# This YAML file contains all API objects that are necessary to run Cinder CSI
+# driver.
+#
+# In production, this needs to be in separate files, e.g. service account and
+# role and role binding needs to be created once.
+#
+# It server as an example how to use external attacher and external provisioner
+# images shipped with {product-title} with a community CSI driver.
+
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: cinder-csi-role
+rules:
+  - apiGroups: [""]
+    resources: ["persistentvolumes"]
+    verbs: ["create", "delete", "get", "list", "watch", "update", "patch"]
+  - apiGroups: [""]
+    resources: ["events"]
+    verbs: ["create", "update", "patch"]
```

Comment:
Still need "list" verb here or we'll get:
E0522 03:32:42.219524       1 controller.go:769] Error watching for provisioning success, can't provision for claim "mytest/pvc1": events is forbidden: User "system:serviceaccount:csi:cinder-csi" cannot list events in the namespace "mytest": User "system:serviceaccount:csi:cinder-csi" cannot list events in project "mytest"

Reply:
added, thanks for spotting this